### ISSUE TITLE
Self-hosted version ask user to login and provide API key. Completion/chat not working.

### ISSUE THREAD

**Issue by hazratisulton** [smallcloudai/refact-intellij#30]
There is several problems.
1.
Self-hosted version trying to login and check API key even if Inference URL set to http://127.0.0.1:8008/)
This messages are repeated in the log:
```
2023-09-08 01:49:25,452 [ 451242]   WARN - check_login - call
2023-09-08 01:49:25,526 [ 451316]   WARN - check_login - recall: {
    "retcode": "FAILED",
    "human_readable_message": "The API key is not recognized (1)."
}
```
Steps to reproduce:
1. Install Refact plugin.
2. Click to login to RefactAI.
3. Click to Settings
4. Set Advanced - Inference URL to http://127.0.0.1:8008/ and save settings.
5 Try completion - it works. Try open chat - No chats availiable

Workaround: login to cloud, then delete API key and set Inference URL for self-hosted.

2.
Sometimes you are not logged as self-hosted but logged as null. In this case completion and chat not working. 
Steps to reproduce: First installation?

Plugin version 1.1.53

**Comment by hazratisulton** [comment]
1.1.53-fixes-9-b294d942 - OK

**Comment by hazratisulton** [comment]
First case fixed here https://plugins.jetbrains.com/plugin/20647-refact-ai/versions/stable/399313
2nd can't reproduce.
